RebelAllianceBank/Classes/FileHandler.cs

```diff
@@ -0,0 +1,184 @@
+        /// <returns>An object of <see cref="IBankAccount"/> or null if row is invalid.</returns>
+        public IBankAccount StoredBankAccount(string[] row)
+        {
+            switch (row[0])
+            {
+                case "0":
+                    return new CardAccount
+                    {
+                        ID = Convert.ToInt32(row[0]),
+                        UserId = row[1],
+                        AccountType = Convert.ToInt32(row[2]),
+                        AccountName = row[3],
+                        Balance = Convert.ToDecimal(row[4]),
+                        AccountCurrency = row[5]
+                    };
+                case "1":
+                    return new SavingsAccount
+                    {
+                        ID = Convert.ToInt32(row[0]),
+                        UserId = row[1],
+                        AccountType = Convert.ToInt32(row[2]),
+                        AccountName = row[3],
+                        Balance = Convert.ToDecimal(row[4]),
+                        AccountCurrency = row[5]
+                    };
+                case "2":
+                    return new ISK
+                    {
+                        ID = Convert.ToInt32(row[0]),
+                        UserId = row[1],
+                        AccountType = Convert.ToInt32(row[2]),
+                        AccountName = row[3],
+                        Balance = Convert.ToDecimal(row[4]),
+                        AccountCurrency = row[5]
+                    };
+                default:
+                    return null;
+            }
+        }
+        /// <summary>
+        /// Writes a bank account to the file and returns the created <see cref="IBankAccount"/> object.
+        /// </summary>
+        /// <param name="id">The ID of the account.</param>
+        /// <param name="accType">The type of the account.</param>
+        /// <param name="name">The name of the account.</param>
+        /// <param name="balance">The balance of the account.</param>
+        /// <param name="curr">The currency of the account.</param>
+        /// <returns>The created <see cref="IBankAccount"/> object or null if writing fails.</returns>
+        public IBankAccount WriteAccountToFile(int id, int accType, string name, decimal balance, string curr)
+        {
+            string filePath = "..\\..\\..\\accounts.txt";
+            if (File.Exists(filePath))
+            {
+                using (StreamWriter sw = new StreamWriter(filePath, append: true))
+                {
+                    string write = $"{id}-{accType}-{name}-{balance}-{curr}";
+                    sw.WriteLine(write, Environment.NewLine);
+                }
+            }
+            IBankAccount account = StoredBankAccount(new string[] { id.ToString(), accType.ToString(), name, balance.ToString(), curr });
+            if (account != null)
+            {
+                return account;
+            }
+            return null;
+        }
```

Comment:
Why does it return an bankaccount? Is the returned object used for anything useful?

Reply:
same as other